lib/post.tsx

```diff
@@ -3,69 +3,95 @@ import fs, { readFileSync } from 'fs';
 import matter from 'gray-matter';
 import { remark } from 'remark';
 import html from 'remark-html';
-//dangerouslySetInnerHTMLがうまく機能しないため下記を追加
-// import remarkParse from 'remark-parse';
-// import { unified } from 'unified';
+import { client } from '../lib/microCMS';
 
 //mdファイルが格納されたディレクトリのパス取得
 const postsDirectory = path.join(process.cwd(), 'posts');
 
-//mdファイルのデータを取得(トップページ表示用)
-export function getPostsData() {
-  const fileNames = fs.readdirSync(postsDirectory);
-  const allPostsData = fileNames.map((fileName) => {
-    const id = fileName.replace(/\.md$/, ''); //ファイル名(URLのid)
+//トップページ用データ取得
+export const getPostsData = async (id) => {
+  const blog = await client.get({
+    endpoint: 'blogs',
+    queries: {
+      ids: id,
+    },
+  });
 
-    //マークダウンファイルを文字列として読み取る
-    const fullPath = path.join(postsDirectory, fileName);
-    const fileContents = readFileSync(fullPath, 'utf8');
-    //メタデータの取得
-    const matterResult = matter(fileContents);
+  return blog;
 
-    //idとデータを返す
-    return {
-      id,
-      ...matterResult.data,
-    };
-  });
-  return allPostsData;
-}
+  //mdファイルのデータを取得(トップページ表示用)
+  // const fileNames = fs.readdirSync(postsDirectory);
+  // const allPostsData = fileNames.map((fileName) => {
+  //   const id = fileName.replace(/\.md$/, ''); //ファイル名(URLのid)
+  //   //マークダウンファイルを文字列として読み取る
+  //   const fullPath = path.join(postsDirectory, fileName);
+  //   const fileContents = readFileSync(fullPath, 'utf8');
+  //   //メタデータの取得
+  //   const matterResult = matter(fileContents);
+  //   //idとデータを返す
+  //   return {
+  //     id,
+  //     ...matterResult.data,
+  //   };
+  // });
+  // return allPostsData;
+};
 
 //getStaticPathsで使用する動的ルーティングのURLを取得
-export function getAllPostIds() {
-  const fileNames = fs.readdirSync(postsDirectory);
-  return fileNames.map((fileName) => {
-    return {
-      params: {
-        id: fileName.replace(/\.md$/, ''), //ファイル名(URLのid)
-      },
-    };
+export async function getAllPostIds() {
+  const blog = await client.get({
+    endpoint: 'blogs',
   });
+  const ids = blog.contents.map((content) => content.id);
+  return ids;
```

Comment:
[IMO]
私だったらまとめちゃいます〜

```suggestion
  return blog.contents.map((content) => content.id);
```